relay-common/src/serde.rs

```diff
@@ -0,0 +1,70 @@
+use serde::ser::Serializer;
+use serde::{Deserialize, Serialize};
+
+/// A wrapper type that is able to deserialize its inner value from a nested JSON string.
+///
+/// For example, this struct:
+///
+/// ```rust
+/// use relay_common::as_string;
+/// use serde::Deserialize;
+///
+/// #[derive(Deserialize)]
+/// struct TraceContext {
+///     #[serde(with = "as_string")]
+///     sample_rate: f64,
+/// }
+/// ```
+///
+/// ...deserializes from `{"sample_rate": "1.0"}` and `{"sample_rate": 1.0}`, and serializes to
+/// `{"sample_rate": "1.0"}`
+pub mod as_string {
+    use super::*;
+
+    /// Deserialize a value from a nested json string
+    pub fn deserialize<'de, T, D>(deserializer: D) -> Result<T, D::Error>
+    where
+        for<'de2> T: Deserialize<'de2>,
+        D: serde::Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        #[serde(untagged)]
+        enum Helper<T> {
+            Verbatim(T),
```

Comment:
Please remove the verbatim case, let's be strict here and just support the string. 

Also, let's move this directly into `relay-sampling` until we have a need to share this with other crates. There is a _similar_ function in Profiling already (see [here](https://github.com/getsentry/relay/blob/e5c80898363871a256073c511bc9ad8b17385e13/relay-server/src/utils/profile.rs#L37)), however it uses `parse()` rather than `serde_json` and it allows both deserializations, so you can't share the implementation.

Let's please keep this as simple as possible.